app/Actions/Search/PhotoSearch.php

```diff
@@ -2,47 +2,35 @@
 
 namespace App\Actions\Search;
 
-use App\Actions\Albums\Extensions\PublicIds;
-use App\Facades\AccessControl;
-use App\Models\Configs;
+use App\Actions\PhotoAuthorisationProvider;
 use App\Models\Photo;
 use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Collection;
 
 class PhotoSearch
 {
-	private function unsorted_or_public(Builder $query): Builder
-	{
-		if (AccessControl::is_admin()) {
-			return $query->orWhere('album_id', '=', null);
-		}
-
-		if (AccessControl::can_upload()) {
-			$query = $query->orWhere(fn ($q) => $q->where('album_id', '=', null)->where('owner_id', '=', AccessControl::id()));
-		}
+	protected PhotoAuthorisationProvider $photoAuthorisationProvider;
 
-		if (Configs::get_value('public_photos_hidden', '1') === '0') {
-			$query = $query->orWhere('public', '=', 1);
-		}
-
-		return $query;
+	public function __construct()
+	{
+		$this->photoAuthorisationProvider = resolve(PhotoAuthorisationProvider::class);
```

Comment:
Same, should use DDI in the constructor.

Reply:
Fixed it.